default.nix

```diff
@@ -38,29 +38,28 @@ let haskellPackages =
                 {
                     streamly-process = mkPackage super "streamly-process" ./. flags inShell;
 
-                    # TODO: Uncomment the following after the streamly release
+                    streamly =
+                      nixpkgs.haskell.lib.overrideCabal
+                        (super.callHackageDirect
+                          { pkg = "streamly";
+                            ver = "0.8.0";
+                            sha256 = "0vy2lkljizlhpbpbybmg9jcmj2g4s1aaqd2dzy5c0y0n4rgwxask";
+                          } {})
+                        (old:
+                          { librarySystemDepends =
+                              if builtins.currentSystem == "x86_64-darwin"
+                              then [nixpkgs.darwin.apple_sdk.frameworks.Cocoa]
+                              else [];
+                            enableLibraryProfiling = false;
+                            doHaddock = false;
+                          });
```

Comment:
`nix-shell --run "cabal build"` will fail to compile.

Reply:
Need to get streamly master

Reply:
In fact without this change cabal build was not working inside nix-shell. With this change it is working. Not sure why it does not work for you.

Reply:
Just ran it again:

```
$ nix-shell --run "cabal build"
Up to date
```

Reply:
I did not use `cabal.project` and `default.nix` at the same time.